test/test_types.js

```diff
@@ -3556,6 +3556,41 @@ suite('types', function () {
       assert.throws(function () { t.encode('hi', buf, -1); });
     });
 
+    ['fooWrapped', 'wrapped.foo'].forEach(function(name) {
+
+      test('nested schema with registry: ' + name, function() {
+        var a = {
+          name: name,
+          type: 'record',
+          fields: [
+            { name: 'a', type: 'boolean'},
+            { name: 'b', type: 'foo'}
```

Comment:
The issue here is that the type `foo` is referenced unqualified, so according to [Avro resolution rules](https://avro.apache.org/docs/1.8.2/spec.html#names) it inherits the namespace of the outer type. When you use `wrapped.foo` as name for the outer type, this field therefore becomes namespaced under `wrapped` as well (i.e. becomes `wrapped.foo`) and ends up referencing the outer type.

One way to achieve what you're trying to do is to prefix the reference with a dot (`.foo`). In that case it will always refer to the registry's `foo` entry.

Reply:
@mtth  Thanks for the clarification, I learned something new today. Thanks for the awesome lib.